### PR TITLE
Documentation for Lie groups, SO(n), SE(n)

### PR DESCRIPTION
**Pull request by stefanheyder**
See also #398.

### CODE REVIEW

File: geomstats/geometry/special_euclidean.py [hunk 1]
```diff
@@ -39,6 +39,21 @@ class SpecialEuclidean(LieGroup):
...
     def __init__(self, n, point_type=None, epsilon=0.):
+        """Initiate an object of class SpecialEuclidean.
+
+        Parameter
+        ---------
+        n : int
+            the dimension of the euclidean space that SE(n) acts upon
+        point_type : str, {'vector', 'matrix'}, optional
+            whether to represent elmenents of SE(n) by vectors or matrices
+            if None is given, point_type is set to 'vector for dimension 3
```
Comment (ninamiolane) [inline geomstats/geometry/special_euclidean.py]: NIT: 'vector -> 'vector' and matrix -> 'matrix'

File: geomstats/geometry/lie_group.py [hunk 2]
```diff
@@ -83,7 +88,12 @@ def get_identity(self, point_type=None):
 
         Parameters
         ----------
-        point_type: str, optional
+        point_type : str, {'matrix', 'vector'}, optional
+            default: the default point type
```
Comment (nguigs) [inline geomstats/geometry/lie_group.py]: indent problem here

File: geomstats/geometry/lie_group.py [hunk 3]
```diff
@@ -297,7 +312,8 @@ def log_not_from_identity(self, point, base_point, point_type):
         ----------
         point : array-like, shape=[n_samples, {dimension,[n,n]}]
         base_point : array-like, shape=[n_samples, {dimension,[n,n]}]
-        point_type : str, {'vector', 'matrix'}
+        point_type : str, {'vector', 'matrix'}, optional
+            defaults to the default point type
```
Comment (nguigs) [inline geomstats/geometry/lie_group.py]: unify specification of the default point_type, `defaults to` or `default:`

File: geomstats/geometry/special_euclidean.py [hunk 4]
```diff
@@ -39,6 +39,21 @@ class SpecialEuclidean(LieGroup):
...
+        Parameter
+        ---------
+        n : int
+            the dimension of the euclidean space that SE(n) acts upon
+        point_type : str, {'vector', 'matrix'}, optional
+            whether to represent elmenents of SE(n) by vectors or matrices
+            if None is given, point_type is set to 'vector' for dimension 3
+            and 'matrix' otherwise
+        epsilon : float, optional
+            precision to use for calculations involving potential divison by in
```
Comment (nguigs) [inline geomstats/geometry/special_euclidean.py]: typo: division by in

File: geomstats/geometry/special_euclidean.py [hunk 5]
```diff
@@ -121,13 +167,14 @@ def regularize_tangent_vec_at_identity(
 
         Parameters
         ----------
-        tangent_vec
-        metric
-        point_type
+        tangent_vec: array-like, shape=[n_samples, {dimension, [n + 1, n + 1]}]
```
Comment (nguigs) [inline geomstats/geometry/special_euclidean.py]: space before colon

File: geomstats/geometry/special_euclidean.py [hunk 6]
```diff
@@ -141,14 +188,16 @@ def regularize_tangent_vec(
 
         Parameters
         ----------
-        tangent_vec
-        base_point
-        metric
-        point_type
+        tangent_vec: array-like, shape=[n_samples, {dimension, [n + 1, n + 1]}]
```
Comment (nguigs) [inline geomstats/geometry/special_euclidean.py]: space before colon

File: geomstats/geometry/special_euclidean.py [hunk 7]
```diff
@@ -141,14 +188,16 @@ def regularize_tangent_vec(
...
+        base_point : array-like, shape=[n_samples, {dimension, [n + 1, n + 1]}]
+        metric : RiemannianMetric, optional
+            defaults to left_canonical_metric
+        point_type: str, {'vector', 'matrix'}, optional
```
Comment (nguigs) [inline geomstats/geometry/special_euclidean.py]: space before colon

File: geomstats/geometry/special_euclidean.py [hunk 8]
```diff
@@ -291,6 +361,22 @@ def jacobian_translation(
 
         Compute the jacobian matrix of the differential
         of the left/right translations from the identity to point in SE(n).
+        Currently only implemented for point_type == 'vector'.
+
+        Parameters
+        ----------
+        point: array-like, shape=[n_samples, {dimension, [n + 1, n + 1]}]
+
+        left_or_right: str, {'left', 'right'}, optional
```
Comment (nguigs) [inline geomstats/geometry/special_euclidean.py]: space before colon

File: geomstats/geometry/special_euclidean.py [hunk 9]
```diff
@@ -348,7 +434,20 @@ def jacobian_translation(
...
+
+        Parameters
+        ----------
+        tangent_vec: array-like, shape=[n_samples, {dimension, [n + 1, n + 1]}]
+        point_type: str, {'vector', 'matrix'}, optional
+            defaults to default_point_type
+
+        Returns
+        -------
+        group_exp: array-like, shape=[n_samples, {dimension, [n + 1, n + 1]}]
```
Comment (nguigs) [inline geomstats/geometry/special_euclidean.py]: space before colon

**Pull request merged**